app/controllers/users_controller.rb

```diff
@@ -1,3 +1,29 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    return if @user
+
+    flash[:warning] = t(".not_found")
+    redirect_to root_path
+  end
```

Comment:
```suggestion
  def show; end
```
E đưa code bên trong ra method load_user và dùng before_action